flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java

```diff
@@ -454,11 +479,7 @@ private static boolean isEnumType(TypeInformation<?> type) {
 		 */
 		@PublicEvolving
 		public IntervalJoined<T1, T2, KEY> between(Time lowerBound, Time upperBound) {
-
-			TimeCharacteristic timeCharacteristic =
-				streamOne.getExecutionEnvironment().getStreamTimeCharacteristic();
-
-			if (timeCharacteristic != TimeCharacteristic.EventTime) {
+			if (timeBehaviour != TimeBehaviour.EventTime) {
```

Comment:
It's interesting, we allow users to call `inProcessingTime()` but it will not work because of this. 😅 Maybe we should just document that an interval join is always in event time and don't allow configuring it for now?

Reply:
@aljoscha I was worried about the other places make data stream with processing time configuration but this handle all interval join operations of data stream with event time at default. It's unreasonable that all join operations are defaultly configured with event time and no validation about this. What do you concern about that this don't allow configuring it for now?

Reply:
No concerns. I just found it funny. Let's leave it like this 👌

Reply:
@aljoscha , therefore any comments on other changes?

Reply:
No, I'll merge this now. Good work!